pkg/controller/provider/controller.go

```diff
@@ -157,6 +160,28 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 					_ = r.DB().Close(true)
 				}
 			}
+
+			// Wait for the PV to release
```

Comment:
yeah I think I mention it in one of our talks that its not the best place since its effect all the providers,  i"ll try to move it directly under the ova instead.

Reply:
Another approach that is worth checking is to use a finalizer for the PVC that woild remove the PV

Reply:
Currently I'm encounter some issues with using the finalizer ,especially watching the PVC to catch the delete operation, so I'm putting this aside for now and switching back to the virt-v2v taks, I"ll come bake to it later (I think is the least urgent task from the existing ones).

Reply:
@bkhizgiy sure, getting the right configuration from the output of virt-v2v is definitely more important than removing the PV of the OVA server

Reply:
looking at this thread a bit more carefully, the finalizer should indeed not be placed on the PVC but on the provider - so when the provider is deleted, we'll try to remove its PV and only when the PV is removed, the finalizer is removed and the deletion of the provider can proceed

Reply:
> looking at this thread a bit more carefully, the finalizer should indeed not be placed on the PVC but on the provider - so when the provider is deleted, we'll try to remove its PV and only when the PV is removed, the finalizer is removed and the deletion of the provider can proceed

@bkhizgiy posted #742 with those changes